moto_week3/s3_functions.py

```diff
@@ -0,0 +1,43 @@
+import boto3
+
+
+def get_client():
+    return boto3.client("s3")
+
+
+def list_s3_buckets():
+    s3 = get_client()
+    response = s3.list_buckets()
+    for bucket in response['Buckets']:
+        yield(bucket['Name'])
+
+
+def list_s3_objects(bucket):
+    s3 = get_client()
+    response = s3.list_objects_v2(Bucket=bucket)
+    if response:
+        try:
+            for _object in response['Contents']:
+                yield(_object['Key'])
+        except KeyError:
+            pass

```

Comment:
Try to inform the requester why exception occurred. Not efficient to pass the exception silently.

Reply:
okay. will keep that in mind and update